docs/source/cli-reference/list-options.rst

```diff
@@ -0,0 +1,49 @@
+.. _cli_list_options:
+
+All Available Options
+=====================
+
+This section contains an exhaustive list of all Scancode options, arranged in various sections.
+The sections are as follows:
+
+- Basic Scan Options
+- Core Scan Options
```

Comment:
We probably want to rename `Core Scan Options` and its `rst` file to something like `Extra Options` or `Advanced Options` IMO.

Reply:
Advanced Options sounds great IMO. Also, note that I have used the structure/grouping of options mentioned in both the help text and in the code `help_group=CORE_GROUP`, it'd be okay if I deviate from that?

Reply:
Yes, I suppose following the various groups is fine. I had forgotten that this is how they are structured internally in the code itself.